src/org/opendatakit/briefcase/model/BriefcaseAnalytics.java

```diff
@@ -0,0 +1,182 @@
+
+  /**
+   * Constructor
+   *
+   * <P>This constructor will instantiate 'brsanthu's {@link GoogleAnalytics} class and
+   * check if the user is okay with their behaviour being tracked.</P>
+   * <P>The constructor is private so as to adhere to the rules of the Singleton pattern.</P>
+   */
+  private BriefcaseAnalytics() {
+    config = new GoogleAnalyticsConfig();
+    checkTrackingConsentState();
+    this.googleAnalytics = new GoogleAnalytics(config, GOOGLE_TRACKING_ID);
+  }
+
+  /**
+   * Singleton creator.
+   *
+   * <P>This method (along with the private constructor) will ensure
+   * that only one instance of this class is instantiated.</P>
+   * @return the instantiated Singleton of {@link BriefcaseAnalytics}.
+   */
+  public static BriefcaseAnalytics getInstance() {
+    return instance == null ? new BriefcaseAnalytics() : instance;
+  }
+
+  /**
+   *  Sends usage information to Google Analytics, only if the user has consented.
+   *  The information is modeled using an {@link EventHit} object.
+   *
+   * @param action (required) name of the event itself.
+   * @param label (optional) an open-ended label.
+   * @param value (optional) an open-ended value. Must be an integer.
+   * @return the {@link EventHit} object.
+   */
+  public EventHit trackEvent(String action, String label, Integer value) {
+    checkTrackingConsentState();
+    EventHit eventHit = new EventBuilder()
+      .trackingID(GOOGLE_TRACKING_ID)
+      .clientID(UNIQUE_USER_ID)
+      .dataSource(System.getProperty("os.name"))
+      .applicationName(BuildConfig.NAME)
+      .applicationVersion(BuildConfig.VERSION)
+      .eventCategory(EVENT_CATEGORY)
+      .eventAction(action)
+      .eventLabel(label)
+      .eventValue(value)
+      .build();
+    googleAnalytics.postAsync(eventHit);
+    return eventHit;
+  }
+
+  /**
+   * Track this application's startup event.
+   *
+   * <P>This should only be run once per application execution.</P>
+   * <P>Note that the method will check to see if it has already tracked this session's startup.</P>
+   */
+  public void trackStartup() {
+    if (hasTrackedStartup == false) {
+      trackEvent("Application-Startup", null, null);
+      hasTrackedStartup = true;
+    }
+  }
+
+  /**
+   * Check if the user has set or updated their consent
+   * to ODK tracking their application usage.
+   *
+   * <P>Note that this method should be executed before any event is tracked,
+   * in case the user has unticked the consent checkbox within the session.</P>
+   */
+  private void checkTrackingConsentState() {
+    briefcaseTrackingConsent = BriefcasePreferences.getBriefcaseTrackingConsentProperty();
+    config.setGatherStats(briefcaseTrackingConsent);
+  }
+
+  /**
+   * A helper class used to streamline the creation Events.
+   *
+   * <P>An {@link EventHit} object is used to model the information being tracked.</P>
+   */
+  private static class EventBuilder {
+
+    private String trackingID = null;
```

Comment:
I'll remove them. Once in the past I tried to initialise a variable within a try-catch black and it insisted that I declare the variable as null before the 'try' section, ever since then I've gotten in to the habit of declaring absolutely everything null - I may rethink this :).